src/main/thrift/parquet.thrift

```diff
@@ -237,6 +237,83 @@ struct SizeStatistics {
    3: optional list<i64> definition_level_histogram;
 }
 
+/**
+ * Interpretation for edges of GEOMETRY logical type, i.e. whether the edge
+ * between points represent a straight cartesian line or the shortest line on
+ * the sphere. Please note that it only applies to polygons.
+ */
+enum Edges {
+  PLANAR = 0;
+  SPHERICAL = 1;
+}
+
+/**
+ * A custom WKB-encoded polygon or multi-polygon to represent a covering of
+ * geometries. For example, it may be a bounding box, or an evelope of geometries
+ * when a bounding box cannot be built (e.g. a geometry has spherical edges, or if
+ * an edge of geographic coordinates crosses the antimeridian). In addition, it can
+ * also be used to provide vendor-agnostic coverings like S2 or H3 grids.
+ */
+struct Covering {
+  /** Bytes of a WKB-encoded geometry */
+  1: required binary geometry;
+  /** Edges of the geometry, which is independent of edges from the logical type */
+  2: required Edges edges;
```

Comment:
@jiayuasu and I were discussion how we could support vendor-agnostic coverings like S2 or H3 grids as stated in the comments above here. The geometry is in binary type, which provides sufficient flexibilities. However, how can we know the binary is stored as S3 or H3 or anything else? Should we add another properties here such as type?

Reply:
Perhaps we can add a `3: optional string metadata;` to store arbitrary information? cc @paleolimbot

Reply:
@paleolimbot this sounds like a good idea. How will the `update` function work for the covering (WKT polygon / S2 / H3)? See our POC implementation (https://github.com/apache/parquet-java/pull/2971/files)

Reply:
This new design of `Covering` looks flexible to me. If this is desired, I can change the PR to adopt it.

> How will the update function work for the covering (WKT polygon / S2 / H3)? 

I think implementations can use any kind of representation (geometry, bounding box, long list, etc). We just need to serialize them into thrift objects when the statistics are finalized.

Reply:
@wgtmac @paleolimbot I agree that we should leave `S2` and `H3` parts out for the `kind` field. The only allowed value is `bbox`, and remove the `edges` field. Then the covering field will be identical to GeoParquet 1.1 spec.